db/config.cc

```diff
@@ -1165,4 +1170,74 @@ future<gms::inet_address> resolve(const config_file::named_value<sstring>& addre
     co_return coroutine::exception(std::move(ex));
 }
 
+static future<std::vector<seastar::metrics::relabel_config>> get_relable_from_file(const std::string& name) {
+    return seastar::open_file_dma(name, open_flags::ro).then([](file f) {
+        return f.size().then([f](size_t s) {
+            return do_with(seastar::make_file_input_stream(f), [s](seastar::input_stream<char>& in) {
+                return in.read_exactly(s).then([](temporary_buffer<char> buf) {
+                    auto yaml = YAML::Load(sstring(buf.begin(), buf.end()));
+                    std::vector<seastar::metrics::relabel_config> relabels;
+                    const YAML::Node& realbels_config = yaml["relabel_configs"];
+                    relabels.resize(realbels_config.size());
+                    size_t i = 0;
+                    for (auto it = realbels_config.begin(); it != realbels_config.end(); ++it, i++) {
+                        const YAML::Node& element = *it;
+                        if (element["source_labels"]) {
+                            auto labels = element["source_labels"];
+                            std::vector<std::string> source_labels;
+                            source_labels.resize(labels.size());
+                            size_t j = 0;
+                            for (auto label_it = labels.begin(); label_it !=  labels.end(); ++label_it, j++) {
+                                source_labels[j] = label_it->as<std::string>();
+                            }
+                            relabels[i].source_labels = source_labels;
+                        }
+                        if (element["action"]) {
+                            relabels[i].action = seastar::metrics::relabel_config_action(element["action"].as<std::string>());
+                        }
+                        if (element["replacement"]) {
+                            relabels[i].replacement = element["replacement"].as<std::string>();
+                        }
+                        if (element["target_label"]) {
+                            relabels[i].target_label = element["target_label"].as<std::string>();
+                        }
+                        if (element["separator"]) {
+                            relabels[i].separator = element["separator"].as<std::string>();
+                        }
+                        if (element["regex"]) {
+                            relabels[i].expr = element["regex"].as<std::string>();
+                        }
```

Comment:
Should we complain about an unrecognized element here?

Reply:
I think you're right, I'll throw on unrecognized element